sdk/formrecognizer/azure-ai-formrecognizer/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,244 @@
+form_pages = poller.result()
+
+for idx, content in enumerate(form_pages):
+    print("----Recognizing content from page #{}----".format(idx+1))
+    # process document layout by parsing content in pages
+    for table_idx, table in enumerate(content.tables):
+        # process document tables
+
+    for line_idx, line in enumerate(content.lines):
+        # process lines
+        for word in line.words:
+            # process words in a line
+
+    for selection_mark in content.selection_marks:
+        # process selection marks that were found
+```
+
+
+Analyzing document layout with `3.2.x`:
+```python
+with open(path_to_sample_documents, "rb") as f:
+    poller = document_analysis_client.begin_analyze_document(
+        "prebuilt-layout", document=f
+    )
+result = poller.result()
+
+for idx, style in enumerate(result.styles):
+    # process document styles
+
+for page in result.pages:
+    # process pages
+    for line in page.lines:
+        # process lines
+    for word in page.words:
+        # process words
+
+for table_idx, table in enumerate(result.tables):
+    # process tables found in document
+```
+
+Analyzing general prebuilt document types with `3.2.x`:
+```python
+with open(path_to_sample_documents, "rb") as f:
+    poller = document_analysis_client.begin_analyze_document(
+        "prebuilt-document", document=f
+    )
+result = poller.result()
+
+for entity in result.entities:
+    # process entities found in document
+
+for kv_pair in result.key_value_pairs:
+    # process key-value pairs found in document
+
+for page in result.pages:
+    # process pages
+    for line in page.lines:
+        # process lines
+    for word in page.words:
+        # process words
+
+for table_idx, table in enumerate(result.tables):
+    # process tables found in document
+```
+
+NOTE: All of these samples also work with `begin_analyze_document_from_url` when providing a valid URL to the document.
+
+### Analyzing a custom model
+
+Differences between the versions:
+- Analyzing a custom model with `DocumentAnalysisClient` uses the general `begin_analyze_document` and `begin_analyze_document_from_url` methods.
+- In order to analyze a custom model with `FormRecognizerClient` the `begin_recognize_custom_models` and its corresponding URL methods are used.
+- The `include_field_elements` kwarg does not exist with the `DocumentAnalysisClient`, text details are automatically included with service version `2021-09-30-preview`.
+
+Analyze custom document with `3.1.x`:
+```python
+with open(path_to_sample_forms, "rb") as f:
+    poller = form_recognizer_client.begin_recognize_custom_forms(
+        model_id=model_id, form=f, include_field_elements=True
+    )
+forms = poller.result()
+```
+
+Analyze custom document with `3.2.x`:
+```python
+with open(path_to_sample_documents, "rb") as f:
+    poller = document_analysis_client.begin_analyze_document(
+        model=model_id, document=f
+    )
+result = poller.result()
+```
+
+### Training a custom model
```

Comment:
FYI that I used training here since this guide is for users who are using the old version and moving to the new one. Within the text I do mention build, but this was on purpose. 

Also, I'm aware that I used analyzing above (I felt that one was not to far a leap from recognizing), but let me know if I should unify terms to those used in a certain version